File: gaia-sdk-javascript/src/graphql/request/type/BehaviourExecutionDetail.ts

import {BehaviourNodeExecution} from "./BehaviourNodeExecution";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Represents a detailed summary of executed entities to a given processInstanceId
 */
export class BehaviourExecutionDetail extends Array<(_:VariableRegistry) => string> {
public _typeName = "BehaviourExecutionDetail";
    public processInstanceId = () => { 
        this.push(_ => "processInstanceId")
    };

    public identityId = () => { 
        this.push(_ => "identityId")
    };

    public qualifier = () => { 
        this.push(_ => "qualifier")
    };

    public behaviours = () => { 
        this.push(_ => "behaviours")
    };

    public behaviourId = () => { 
        this.push(_ => "behaviourId")
    };

    public nodes = (config: (_:BehaviourNodeExecution) => void) => this.push((registry) => {
        const entity = new BehaviourNodeExecution();
        config(entity);
        return "nodes { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/BehaviourExecutionDetail.py

from gaia_sdk.graphql.request.type.BehaviourNodeExecution import BehaviourNodeExecution

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class BehaviourExecutionDetail(list):
    """
    Represents a detailed summary of executed entities to a given processInstanceId
    """

    def process_instance_id(self):
        self.append(lambda x: "processInstanceId")

    def identity_id(self):
        self.append(lambda x: "identityId")

    def qualifier(self):
        self.append(lambda x: "qualifier")

    def behaviours(self):
        self.append(lambda x: "behaviours")

    def behaviour_id(self):
        self.append(lambda x: "behaviourId")

    def nodes(self, config: Callable[['BehaviourNodeExecution'], None]):
        def callback(registry: VariableRegistry):
            entity = BehaviourNodeExecution()
            config(entity)
            return "nodes {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
